include/real/exact_number.hpp

```diff
@@ -879,6 +879,47 @@ namespace boost {
                 (*this).normalize();
             }
 
+            /*              BINARY EXPONENTIATION
+             *  @brief:  calculates exact_number^exact_number, (only integral powers)
+             *  @params: number: an exact_number whose integral power is to be evaluated
+             *  @params: exponent: an exact_number which is integer power to be evaluated
+             *  @return: returns an exact_number = number^exponent
+             */
+
+
+            static exact_number<T> binary_exponentiation(const exact_number<T>& number, const exact_number<T>& exponent) {
+                exact_number<T> result, zero = exact_number<T> (), number_copy;
```

Comment:
I suggest we should change exponent to be a "integer_number" type. We have % operator implemented for it also, the division system when we expected result to be an integer for division of integers, returning floor of result in integer form. Like 5/2 = 2.
You can look at line 467, in "real_rational.hpp" file in my current PR.

Reply:
I thought about this, but since your PR wasn't merged yet, I couldn't use integer_number. When its merged I'll change this.

Reply:
Okay, that's fine.